### DISCUSSION TITLE
Databend adapter feasibility

### DISCUSSION THREAD

**Discussion by hanxuanliang** [tconbeer/harlequin#557] [Q&A, Troubleshooting, Support]
Hi @tconbeer!

[Databend](https://github.com/datafuselabs/databend) is a modern alternative to Snowflake.  Recently, I added its adapter to Harlequin. However, I'm not sure how to test its availability in Harlequin while also releasing it so that other developers can use it as well.

The repo is [here](https://github.com/hanxuanliang/harlequin-databend)

Thank you very much for your help!

**Comment by tconbeer** [comment]
Here is the section of the guide about packaging:
https://harlequin.sh/docs/contributing/adapter-guide#packaging-and-distributing-adapters

You have done the first part, making your package discoverable as a plug-in. (Although I recommend a change, see below).

You haven't done the second part, making your package installable as an extra. You can open a PR against this repo to do that.

re:  "releasing it so that other developers can use it as well": you can publish your package to PyPI (lots of good tutorials on this, poetry makes it easy), or just use the git url, and people can install from that.

I'd also recommend taking the steps in the [documenting adapters](https://harlequin.sh/docs/contributing/adapter-guide#documenting-adapters) section, and opening a docs PR against the [docs repo](https://github.com/tconbeer/harlequin-web) with changes that add databend.

Looking at your pyproject.toml file, I would change this line:
https://github.com/hanxuanliang/harlequin-databend/blob/fd1ea3aa534129c853fd8216a436673b1ef9ba9c/pyproject.toml#L13

To be:

```toml
databend = "harlequin_databend:HarlequinDatabendAdapter"
```

The key (on the left side of the `=` in the toml file) determines the adapter "name" used by Harlequin, so after the change above, someone could use this adapter with `harlequin -a databend`; without that change, they would need to specify `harlequin -a 'databend-adapter'`

**Reply by hanxuanliang** [reply]
That's great. I used poetry to publish the databend plugin. Can I test the `harlequin -a databend` directly locally?

**Reply by hanxuanliang** [reply]
The databend plugin is here: [harlequin-databend](https://pypi.org/project/harlequin-databend/).

I will conduct a local test and submit a pull request to ensure its functionality.

**Comment by hanxuanliang** [comment]
![image](https://github.com/tconbeer/harlequin/assets/29874640/f69d5c0d-ef18-4d20-8765-b7abf6730c24)

now it looks fine. I ready to give a pull request, and there may still be some problems, I will continue to update this plugin
